### PR TITLE
Fix for bad hex strings when using `Number` constructor.

### PR DESCRIPTION
**Pull request by Xotic750**
Ref https://github.com/paulmillr/es6-shim/pull/366

Node v0.8, old Chrome/Chromium and others.

### CODE REVIEW

**Comment by ljharb** [comment]
Can you please provide tests that would fail without your change?

Also, can you link to spec text that says that `'-0x1'` should be `NaN`, when the literal `-0x1` parses fine?

**Comment by Xotic750** [comment]
I already provided the tests in https://github.com/paulmillr/es6-shim/pull/366 and that was the only failing test remaining in your latest push.

**Comment by ljharb** [comment]
You're right about the tests - https://travis-ci.org/paulmillr/es6-shim/jobs/89172994 passes but https://travis-ci.org/paulmillr/es6-shim/jobs/88743727 fails.

It's worth noting that this appears to make things consistent, but I do want to know what the spec requires.

**Comment by Xotic750** [comment]
i would guess these
http://www.ecma-international.org/ecma-262/6.0/#sec-tonumber-applied-to-the-string-type
> StrNumericLiteral :::
    StrDecimalLiteral
    BinaryIntegerLiteral
    OctalIntegerLiteral
    HexIntegerLiteral

http://www.ecma-international.org/ecma-262/6.0/#sec-number-conversions

> HexIntegerLiteral :::
    0x HexDigit
   0X HexDigit
   HexIntegerLiteral HexDigit

So, current environments seem to interpret this as `'0x1'` is valid but `'-0x1'` is invalid.

File: es6-shim.js [hunk 1]
```diff
@@ -719,6 +719,7 @@
   };
   var nonWS = ['\u0085', '\u200b', '\ufffe'].join('');
   var nonWSregex = new RegExp('[' + nonWS + ']', 'g');
+  var isBadHexRegex = new RegExp('^[-+]0x[0-9a-f]+$', 'i');
```
Comment (ljharb) [inline es6-shim.js]: This can just be a literal - `/^[\-+]0x[0-9a-f]+$/i`

Comment (Xotic750) [inline es6-shim.js]: I wasn't sure on the preference of style.

Comment (ljharb) [inline es6-shim.js]: In every language, using literals is preferred over more verbose forms when they're equivalent :-)

**Comment by ljharb** [comment]
Please file a separate issue for `parseInt`

File: es6-shim.js (hunk 1 above)
Comment (Xotic750) [inline es6-shim.js]: Understood, but I seem to remember some older environments had problems with certain literals, I think it was `XRegExp` that pointed these out. :)

**Pull request merged**